File: sdk/typescript/src/serialize-payload.ts
/**
 * Unified transport outbound payload serialization.
 */

import { buildOutboundEnvelope, serializeEnvelope } from "./envelope.js";
import type { OutboundWireFormat, SerializeOutboundParams } from "./types.js";

/**
 * Serialize Agent reply for wire transport.
 */
export function serializeForTransport(params: SerializeOutboundParams): string {
  const format: OutboundWireFormat = params.format ?? "envelope";

  if (format === "plainText") {
    return params.text;
  }

  if (format === "legacyJsonText") {
    return JSON.stringify({ text: params.text });
  }

  const envelope = buildOutboundEnvelope({
    channel: params.channel,
    accountId: params.accountId,
    userId: params.userId,
    text: params.text,
    agentId: params.agentId,
    headers: {
      ...params.headers,
      ...(params.replyRoute ? { replyRoute: params.replyRoute } : {}),
    },
  });
  return serializeEnvelope(envelope);
}

File: sdk/python/openclaw_message_sdk/serialize_payload.py
"""Unified transport outbound payload serialization."""

from __future__ import annotations

import json

from openclaw_message_sdk.envelope import build_outbound_envelope, serialize_envelope
from openclaw_message_sdk.types import OutboundWireFormat, SerializeOutboundParams


def serialize_for_transport(params: SerializeOutboundParams) -> str:
    """Serialize Agent reply for wire transport."""
    wire_format: OutboundWireFormat = params.get("format", "envelope")
    text = params["text"]

    if wire_format == "plainText":
        return text

    if wire_format == "legacyJsonText":
        return json.dumps({"text": text}, ensure_ascii=False)

    headers = dict(params.get("headers") or {})
    reply_route = params.get("replyRoute")
    if reply_route:
        headers["replyRoute"] = reply_route

    envelope = build_outbound_envelope(
        channel=params["channel"],
        account_id=params["accountId"],
        user_id=params["userId"],
        text=text,
        agent_id=params.get("agentId"),
        headers=headers or None,
    )
    return serialize_envelope(envelope)
